src/app-layer-htp.c

```diff
@@ -2779,11 +2780,11 @@ static void HTPConfigParseParameters(HTPCfgRec *cfg_prec, ConfNode *s,
                 cfg_prec->randomize = ConfValIsTrue(p->val);
             }
         } else if (strcasecmp("randomize-inspection-range", p->name) == 0) {
-            uint32_t range = atoi(p->val);
-            if (range > 100) {
-                SCLogError(SC_ERR_SIZE_PARSE, "Invalid value for randomize"
-                           " inspection range setting from conf file - %s."
-                           " It should be inferior to 100."
+            uint32_t range;
+            if (StringParseUint32wValidation(&range, 10, 0, (const char *)p->val, 0, 99) < 0) {
+                SCLogError(SC_ERR_INVALID_VALUE, "Invalid value for randomize"
```

Comment:
Since we error here and in `StringParseUint32wValidation` meaning we get 2 error messages per single issue. Maybe leave this one and silence the one in `StringParseUint32wValidation`?

Reply:
Right. OK.